lib/amazonka-core/src/Amazonka/Sign/V4.hs

```diff
@@ -42,7 +50,15 @@ presign ex rq a r ts = Base.signRequest meta mempty auth
         . pair (CI.original hAMZSignedHeaders) (toBS shs)
         . pair (CI.original hAMZToken) (toBS <$> _authSessionToken a)
 
-    digest = Base.Tag "UNSIGNED-PAYLOAD"
+    digest =
+      case _requestBody rq of
+        Chunked _ -> unsignedPayload
+        Hashed (HashedStream h _ _) -> Base.Tag $ BAE.convertToBase BAE.Base16 h
```

Comment:
Use `encodeBase16` from `Amazonka.Bytes`, which I only discovered this morning.

Reply:
FIxed